File: 0014-longest-common-prefix/0014-longest-common-prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        sort(strs.begin(), strs.end());
        string std1 = strs[0];
        string std2 = strs[strs.size()-1];
        string prefix = "";
        for (int i=0; i<std1.size(); i++) {
            if (std1[i]==std2[i]) prefix += std1[i];
            else break;
        }

        return prefix;
    }
};

File: 0014-longest-common-prefix/0014-longest-common-prefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        if len(strs) == 1:
            return strs[0]
        sorted_strs = sorted(strs, key = len)
        shortest = sorted_strs[0]
        length = 0
        while length < len(shortest):
            for word in sorted_strs:
                if shortest[:length+1] != word[:length+1]:
                    return shortest[:length]
            length += 1
        return shortest 